std/stdio.d

```diff
@@ -2902,12 +2903,7 @@ struct LockingTextReader
         auto s = chars;
         _front = decodeFront(s);
 
-        // Put everything back.
-        foreach(immutable i; 0 .. chars.length)
-        {
-            immutable c = chars[$ - 1 - i];
-            enforce(ungetc(c, cast(FILE*) _f._p.handle) == c);
-        }
+        _f.seek(start); // rewind
```

Comment:
This is horrible. Do we ungetc the file only to maintain range abstraction?? I think it's apparent that C's I/O doesn't map well to codepoint ranges so one more point for native i/o package.